common-lib/src/main/scala/com/gu/mediaservice/lib/imaging/ImageOperations.scala

```diff
@@ -79,38 +78,41 @@ class ImageOperations(playPath: String) {
       ).map(_ => sourceFile)
   }
 
-  def resizeImage(sourceFile: File, sourceMimeType: Option[String], dimensions: Dimensions,
-                  qual: Double = 100d, tempDir: File, fileType: String): Future[File] = {
+  def resizeImage(sourceFile: File, sourceMimeType: Option[MimeType], dimensions: Dimensions,
+                  qual: Double = 100d, tempDir: File, fileType: MimeType): Future[File] = {
     for {
-      outputFile  <- createTempFile(s"resize-", s".${fileType}", tempDir)
+      outputFile  <- createTempFile(s"resize-", s".${fileType.fileExtension}", tempDir)
       resizeSource = addImage(sourceFile)
       qualified    = quality(resizeSource)(qual)
       resized      = scale(qualified)(dimensions)
       addOutput    = addDestImage(resized)(outputFile)
-      _           <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains("image/tiff"))
+      _           <- runConvertCmd(addOutput, useImageMagick = sourceMimeType.contains(Tiff))
     }
     yield outputFile
   }
 
-  def optimiseImage(resizedFile: File, mediaType: MimeType): File =
+  def optimiseImage(resizedFile: File, mediaType: MimeType): File = mediaType match {
+    case Png => {
+      val fileName: String = resizedFile.getAbsolutePath()
 
-    mediaType.name match {
-      case "image/png" => {
-        val fileName: String = resizedFile.getAbsolutePath()
+      val optimisedImageName: String = fileName.split('.')(0) + "optimised.png"
+      Seq("pngquant",  "--quality", "1-85", fileName, "--output", optimisedImageName).!
 
-        val optimisedImageName: String = fileName.split('.')(0) + "optimised.png"
-        Seq("pngquant",  "--quality", "1-85", fileName, "--output", optimisedImageName).!
-
-        new File(optimisedImageName)
-
-      }
-      case "image/jpeg" => resizedFile
+      new File(optimisedImageName)
     }
+    case Jpeg => resizedFile
+
+    // This should never happen as we only ever crop as PNG or JPEG. See `Crops.cropType` and `CropsTest`
+    // TODO We should create a `CroppingMimeType` to enforce this at the type level.
+    //  However we'd need to change the `Asset` model as source image and crop use this model
+    //  and a source can legally be a `Tiff`. It's not a small change...
+    case Tiff => resizedFile
```

Comment:
This is a bit contentious, hoping the comment is enough for now.

Reply:
Should we log/throw if we get into this illegal state?

Reply:
Good idea!